shm/player.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <memory>
+
+#include "cargo.hpp"
+#include "ship.hpp"
+#include "timeSHM.hpp"
+
+class Coordinates;
+class Player : public Observer{
+public:
+
+    Player(std::shared_ptr<Ship> ship, size_t money, size_t space, std::shared_ptr<Time> publisher);
+
+    Player(std::shared_ptr<Ship> ship, std::shared_ptr<Time> publisher);
+    
+    ~Player(){
+        this -> publisher_ -> removeObserver(this);
+        std::cout << "Goodbye! I was your Player\n";
+    };
+
+    size_t getMoney() const { return money_; }
+    void   giveMoney(size_t money);
+    void   takeMoney(size_t money);
+    size_t getAvailableSpace();
+    size_t getSpeed() const;
+    void   giveCargo(Alcohol* cargo);
+    void   giveCargo(Fruit* cargo);
+    void   giveCargo(Item* cargo);
+    void   removeAlco (Alcohol* cargo);
+    void   removeFruit(Fruit* cargo);
+    void   removeItem (Item* cargo);
```

Comment:
Redundat. Cargo is interface for Alcohol, Item and Fruit. You always should depend on Cargo*.

Reply:
Cargo* is good when you want to access basic values like name,but what if we would like to access expiryDate from Fruit-type item. Can we access it when we by Cargo* object?